csharp/extractor/Semmle.Extraction.CSharp.Standalone/AssemblyCache.cs

```diff
@@ -55,7 +57,9 @@ private void IndexReferences()
 
             // Index "assemblyInfo" by version string
             // The OrderBy is used to ensure that we by default select the highest version number.
-            foreach (var info in assemblyInfoByFileName.Values.OrderBy(info => info.Id))
+            foreach (var info in assemblyInfoByFileName.Values
+                .OrderBy(info => info.Name)
+                .ThenBy(info => info.Version ?? emptyVersion))
```

Comment:
`AssemblyInfo.Id` is implemented as `{name}, Version={version}`. The previous implementation was doing a string based sorting which would not work correctly for versions. For example: `name, Version=9.0.0.0` vs `name, Version=10.0.0.0`.